docs/contribute/guidelines.md

```diff
@@ -0,0 +1,269 @@
+---
+title: Coding and Commit Guidelines
+---
+
+# {{ page.title }}
+{:.no_toc}
+
+{% include toc.html %}
+
+## Git
+
+### Configuration
+1.  Make sure that both "Author" and "Committer" conform to the format:
+
+        John Doe <john.doe@anywhere.com>
+
+    The following formats are **not** allowed: "**johnd**
+    &lt;john.doe@...&gt;" or "**johnny** &lt;john.doe@...&gt;" or
+    "**John** &lt;john.doe@...&gt;"
+
+
+2.  Make sure you commit files with Unix line endings. Refer to the [Git
+    manual](https://git-scm.com/book/tr/v2/Customizing-Git-Git-Configuration#Formatting-and-Whitespace)
+    for more information.
+
+### Commit message
+
+1.  Follow the general rule of how Git commit messages are formatted.
+    Refer to the [Git manual](https://git-scm.com/book/ch5-2.md#Commit-Guidelines) for
+    more information. In particular, every patch must have an
+    informative short summary or "subject line" (also see next point in
+    the list) and it should have a more detailed explanation below.
+2.  In addition, use one of the following **type tags** in the subject
+    line to make it easier for people to understand what your commit was
+    about:
+
+* `[NOP]` - This commit did not have any effect and only concerns whitespace, removing unused methods, fixing documentation typos, etc.
```

Comment:
Needs to be a table

Reply:
I will do this, but I wil also add a TODO, because this will lead to a less readable table, because the default width of the `main_content` div is to small.

Reply:
The TODO is not necessary the table looks good. It seems like as if my styling or formatting changes solved the problem.